Загрузка проекта

pligina-e opened the pull request:
Patch-запрос так и не удалось осуществить, поэтому в классе UpdateEntityTest много ненужного кода. Сделано еще два доп. задания, третье доп. задание в работе

pligina-e committed:
Загрузка проекта

Patch-запрос так и не удалось осуществить, поэтому в классе UpdateEntityTest много ненужного кода. Сделано еще два доп. задания, третье доп. задание в работе

pligina-e committed:
Update README.md

pligina-e committed:
Update testng.xml

pligina-e committed:
Update BaseTest.java

pligina-e committed:
Update CreateEntityTest.java

pligina-e committed:
Update UpdateEntityTest.java

pligina-e committed:
Update Steps.java

pligina-e committed:
Add files

pligina-e committed:
Update README.md

pligina-e commented:
Отчет Allure - параллельный запуск тестов
![image](https://github.com/pligina-e/ApiAutoTest_TestService/assets/82607076/5784a81b-b5e4-4359-bb03-4bce85a172f5)
![image](https://github.com/pligina-e/ApiAutoTest_TestService/assets/82607076/275aef25-7c92-4587-91c1-a43a79479a4f)

avoevodskii commented on src/test/java/test/service/steps/Steps.java:
```diff
@@ -0,0 +1,78 @@
+        return requestSpecification
+                .contentType("application/json")
+            .when()
+                .get("/api/get/" + id)
```
эндпоинт можно вынести в константу

avoevodskii commented on src/test/java/test/service/steps/Steps.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    @Step("Метод для сериализаии. Используется для обновления сущности PATCH-запросом")
+    public EntityAdditionRequestModel getEntityAdditionData(String title, boolean verified, String additional_info, int additional_number, List<Integer> important_numbers) {
```
Это не сериализация) Для упрощения можно передавать сам объект и уже в методе его обрабатывать. Так количество параметров метода уменьшится

pligina-e replied:
Ой, поняла

pligina-e replied:
Не знаю, как это осуществить в данном случае. Только если убрать параметры и вручную в методе прописать значения:
![image](https://github.com/pligina-e/ApiAutoTest_TestService/assets/82607076/667a7ca7-b79d-46f1-b14e-62d150cc00cf)

avoevodskii commented on src/test/java/test/service/tests/BaseTest.java:
```diff
@@ -0,0 +1,19 @@
+import static io.restassured.RestAssured.given;
+
+public class BaseTest {
+    final String GET_ENDPOINT = "/api/get/";
```
Почему эти константы не задействованы в запросах?

pligina-e replied:
Они задействованы в запросах тех классов, которые наследуются от BaseTest

pligina-e committed:
Update Steps.java

pligina-e committed:
Update UpdateEntityTest.java

pligina-e committed:
Update UpdateEntityTest.java

pligina-e committed:
Update Steps.java

pligina-e committed:
Create maven.yml

pligina-e committed:
Update AdditionRequestModel.java

pligina-e committed:
Update EntityAdditionRequestModel.java

pligina-e committed:
Update maven.yml

pligina-e committed:
Update maven.yml

pligina-e committed:
Update GetAllEntitiesTest.java

Обновлен таким образом, потому что при одном первом запуске тестов первая сущность удаляется тестом с запросом Delete, поэтому проверку надо начать  с айди = 2

pligina-e committed:
Update README.md

pligina-e committed:
Update README.md

avoevodskii commented on src/test/java/test/service/tests/DeleteEntityTest.java:
```diff
@@ -0,0 +1,33 @@
+        requestSpecification
+                .header("Content-type", "application/json")
+            .when()
+                .delete("/api/delete/" + idToDeleteEntity)
```
Наследник BaseTest, но константы эндпоинтов все равно проигнорированы

pligina-e committed:
Update DeleteEntityTest.java

pligina-e committed:
Update GetAllEntitiesTest.java

pligina-e commented:
Спасибо за замечания!
